src/components/filters.tsx

```diff
@@ -258,7 +258,34 @@ function buildDirTree(paths: string[], expanded: string[]): Directory {
         }
     });
 
-    return root;
+    return condensedDirTree ? condenseTree(root) : root;
+}
+
+function condenseTree(root: Directory): Directory {
+    const result = getSequenceRoot(root);
+
+    for (const [key, dir] of result.subdirs) {
+        dir.level = result.level + 1;
+        const condensedDir = condenseTree(dir);
+        result.subdirs.set(key, condensedDir);
+    }
+
+    return result;
+}
+
+function getSequenceRoot(root: Directory): Directory {
```

Comment:
This can be an inline lambda inside `compactTree()`. Also maybe `squashSingleChildSubdirs` sounds like a better name.

Reply:
then I think it will de duplicated in each call

`compactTree` is recursive called func.

Reply:
this const will be duplicated in each stack call

<img width="578" alt="Screenshot 2024-01-08 at 01 19 06" src="https://github.com/openscopeproject/TrguiNG/assets/1858880/896d93f5-c7b3-4763-87f7-d02fe8f31254">

Reply:
Fair. In that case remove the commented lambda.